charts/producer-app/templates/job.yaml

```diff
@@ -11,86 +16,16 @@ metadata:
     {{ $key }}: {{ $value }}
     {{- end }}
 spec:
+  suspend: {{ .Values.suspend }}
   {{- if .Values.schedule }}
   schedule: {{ quote .Values.schedule }}
-  {{- end }}
-  suspend: {{ .Values.suspend }}
   successfulJobsHistoryLimit: 1
   concurrencyPolicy: Replace
   jobTemplate:
     spec:
       template:
```

Comment:
Did you test this in the cluster? This is going to be an empty key when no schedule is set